signac/core/syncedattrdict.py

```diff
@@ -194,17 +178,18 @@ def clear(self):
         self.sync()
 
     def update(self, mapping):
```

Comment:
I have also found this on link
>update() accepts either another dictionary object or an iterable of key/value pairs (as tuples or other iterables of length two). If keyword arguments are specified, the dictionary is then updated with those key/value pairs: d.update(red=1, blue=2).

Currently, we support only mapping as input. Should we add iterable of key/value pairs or `kwargs` also?

Reply:
@vishav1771 I'll make a code suggestion.

Reply:
@vishav1771 I can't make a code suggestion inline because it would alter deleted lines, but here's the snippet I would try. You may need to add `from collections.abc import Mapping`.

```python
    def update(self, other=None, **kwargs):
        self.load()
        if other is not None:
            # Convert sequence of key, value pairs to dict before validation
            if not isinstance(other, Mapping):
                other = dict(other)
            self._validate(other)
        if kwargs:
            self._validate(kwargs)
        with self._suspend_sync():
            for key, value in other.items():
                self._data[key] = self.from_base(data=value, parent=self)
            for key, value in kwargs.items():
                self._data[key] = self.from_base(data=value, parent=self)
        self.sync()
```